library/init/category/reader.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+
+The reader monad transformer for passing immutable state.
+-/
+
+prelude
+import init.category.lift init.category.id
+universes u v w
+
+structure reader_t (r : Type u) (m : Type u → Type v) (α : Type u) : Type (max u v) :=
```

Comment:
In this file, `r` is being used to name a type and elements of this type.
For example, `⟨λ r, pure a⟩` is `⟨λ r : r, pure a⟩`. I think this is really confusing.
Could you please use `env` or a greek letter to name `r`?
Or use something different from `r` to name variables of type `r`.

Reply:
I now switched to rho for `reader_t` and `cont_t`